Tests/NostrSDKTests/NIP44v2EncryptingTests.swift

```diff
@@ -0,0 +1,191 @@
+            let conversationKey1Bytes = try conversationKey(
+                privateKeyA: keypair1.privateKey,
+                publicKeyB: keypair2.publicKey
+            ).bytes
+            XCTAssertEqual(expectedConversationKey, Data(conversationKey1Bytes).hexString)
+
+            // Verify payload.
+            let ciphertext = try encrypt(
+                plaintext: plaintext,
+                conversationKey: conversationKey1Bytes,
+                nonce: nonce
+            )
+            XCTAssertEqual(payload, ciphertext)
+
+            // Conversation key from sec2 and pub1.
+            let conversationKey2Bytes = try conversationKey(
+                privateKeyA: keypair2.privateKey,
+                publicKeyB: keypair1.publicKey
+            ).bytes
+            XCTAssertEqual(expectedConversationKey, Data(conversationKey2Bytes).hexString)
+
+            // Verify that decrypted data equals the plaintext that we started off with.
+            let decrypted = try decrypt(payload: payload, conversationKey: conversationKey2Bytes)
+            XCTAssertEqual(decrypted, plaintext)
+        }
+    }
+
+    /// Same as previous step, but instead of a full plaintext and payload, their checksum is provided.
+    func testValidEncryptDecryptLongMessage() throws {
+        let encryptDecryptVectors = try XCTUnwrap(vectors.v2.valid.encryptDecryptLongMessage)
+        try encryptDecryptVectors.forEach { vector in
+            let conversationKey = vector.conversationKey
+            let conversationKeyData = try XCTUnwrap(conversationKey.hexDecoded)
+            let conversationKeyBytes = conversationKeyData.bytes
+
+            let nonce = try XCTUnwrap(vector.nonce.hexDecoded)
+            let expectedPlaintextSHA256 = vector.plaintextSHA256
+
+            let plaintext = String(repeating: vector.pattern, count: vector.repeatCount)
+            let plaintextData = try XCTUnwrap(plaintext.data(using: .utf8))
+            let plaintextSHA256 = plaintextData.sha256
+
+            XCTAssertEqual(plaintextSHA256.hexString, expectedPlaintextSHA256)
+
+            let payloadSHA256 = vector.payloadSHA256
+
+            let ciphertext = try encrypt(
+                plaintext: plaintext,
+                conversationKey: conversationKeyBytes,
+                nonce: nonce
+            )
+            let ciphertextData = try XCTUnwrap(ciphertext.data(using: .utf8))
+            let ciphertextSHA256 = ciphertextData.sha256.hexString
+            XCTAssertEqual(ciphertextSHA256, payloadSHA256)
+
+            let decrypted = try decrypt(payload: ciphertext, conversationKey: conversationKeyBytes)
+            XCTAssertEqual(decrypted, plaintext)
+        }
+    }
+
+    /// Emulate real conversation with only the public encrypt and decrypt functions,
+    /// where the nonce used for encryption is a cryptographically secure pseudorandom generated series of bytes.
+    func testValidEncryptDecryptRandomNonce() throws {
```

Comment:
Added this function so that it's a true test of how the API would be used in the wild.